cookiecutter/prompt.py

```diff
@@ -313,3 +341,36 @@ def prompt_for_config(context, no_input=False):
             raise UndefinedVariableInTemplate(msg, err, context) from err
 
     return cookiecutter_dict
+
+
+def choose_nested_template(context: dict, repo_dir: str, no_input: bool = False) -> str:
+    """Prompt user to select the nested template to use.
+
+    :param context: Source for field names and sample values.
+    :param repo_dir: Repository directory.
+    :param no_input: Do not prompt for user input and use only values from context.
+    :returns: Path to the selected template.
+    """
+    cookiecutter_dict = OrderedDict([])
+    env = StrictEnvironment(context=context)
+    prefix = ""
+    prompts = _extract_prompts(context)
+    key = "templates"
+    config = context['cookiecutter'].get(key, {})
+    if config:
+        # Pass
+        val = prompt_choice_for_template(key, config, no_input)
+        template = config[val]["path"]
+    else:
+        # Old style
+        key = "template"
+        config = context['cookiecutter'].get(key)
```

Comment:
This needs to have a default value, as it does above, to prevent crashing in `prompt_choice_for_config()` under this circumstance:

* `key` is not in `context['cookiecutter']`
* `config` is then `None`
* In `prompt_choice_for_config()`, an attempt is made to iterate over `None`

I recommend adding a test case to demonstrate this issue, and then fix it.